tools/pipeline-generator/Azure.Sdk.Tools.PipelineGenerator/Conventions/UnifiedPipelineConvention.cs

```diff
@@ -92,13 +92,13 @@ protected override async Task<bool> ApplyConventionAsync(BuildDefinition definit
             }
             else
             {
-                if (prTrigger.SettingsSourceType != 2 ||
+                if (prTrigger.SettingsSourceType != 1 ||
```

Comment:
I do not understand why this would fix the issue. Can you point me at more documentation about what this setting means? 

By the way I changed this to 2 because it looked like we were changing all of them to 2 in other places and this one place caused us to change always conflict between 1 here and 2 in the base https://github.com/Azure/azure-sdk-tools/commit/575de41e5357363b2a42fc0aa5f3078a49ee2789#diff-cdd9300bdb4aa1555924639452c0a8d7. Perhaps that isn't an issue any longer since I removed the base. In either case please double check that we don't write anything if we don't change anything I don't want us to always have a dirty state everytime we run the generator.

Reply:
The setting is not actually documented anywhere. I found out about it whilst observing that pipeline definitions with PR triggers would behave one way sometimes, and another way other times. I ended up looking at the REST API traffic and reading through the Azure DevOps source code the figure it out :)

I just double checked that executing it twice didn't cause flip-flopping so I think we should be OK.

I think the summary on the setting is that when it is set to 2, Azure Pipelines interprets the PR trigger one way, and when it is 1 (or nothing) it interprets it another.

Reply:
Actually the code in Azure Pipelines isn't super clear - but I think it has the effect of turning this override on and off.

![image](https://user-images.githubusercontent.com/513398/68637710-9648cf80-0553-11ea-8115-b2808a401da2.png)

Reply:
If that is the case we should audit the other usages and see if they are correct.

Reply:
They are. Type 2 is appropriate for the PR convention. And Type 1 is appropriate for the integration test convention.

Reply:
So the PR validation comment triggering is the only one we want to override?